lstchain/tools/calc_camera_calibration.py

```diff
@@ -0,0 +1,197 @@
+                        ped_data.meta['config'] = self.config
+                        ped_initialized = True
+                    else:
+                        self.log.debug(f"write pedestal data")
+                        # write only after a first event (used to initialize the mask)
+                        self.writer.write('pedestal', ped_data)
+
+            # consider flat field events only after first pedestal event
+            elif event.r0.tel[self.tel_id].trigger_type == 1 and ped_initialized:
+                if self.flatfield.calculate_relative_gain(event):
+
+                    self.log.debug(f"new flatfield at event n. {count+1}, id {event.r0.event_id}")
+
+                    # update the flatfield mask
+                    status_data.flatfield_mask = np.logical_or(ff_data.charge_median_outliers,
+                                                                ff_data.time_median_outliers)
+
+                    calib_data.pixel_status_mask = np.logical_or(status_data.pedestal_mask,status_data.flatfield_mask)
+
+                    masked_charge = np.ma.array(ff_data.charge_median -
+                                                ped_data.charge_median, mask=calib_data.pixel_status_mask)
+
+                    masked_std_square = np.ma.array(ff_data.charge_std ** 2 - ped_data.charge_std ** 2,
+                                                    mask=calib_data.pixel_status_mask)
+                    masked_time = np.ma.array(ff_data.relative_time_median, mask=calib_data.pixel_status_mask)
+
+                    masked_n_phe = 1.2 * masked_charge ** 2 / masked_std_square
+                    masked_n_phe_median = np.ma.median(masked_n_phe)
+
+                    # calculate the calibration data
+                    calib_data.n_phe = np.ma.getdata(masked_n_phe)
+                    calib_data.dc_to_phe = np.ma.getdata(masked_n_phe_median/masked_charge)
+                    calib_data.time_correction = - np.ma.getdata(masked_time)
+
+                    # save the config, to be retrieved as data.meta['config']
+                    if not ff_initialized:
+                        ff_data.meta['config']=self.config
+                        calib_data.meta['config'] = self.config
+                        ff_initialized = True
+                    else:
+                        # write only after a first event (used to initialize the mask)
+                        self.log.debug(f"write flatfield data")
+                        self.writer.write('flatfield', ff_data)
+                        self.log.debug(f"write pixel_status data")
+                        self.writer.write('pixel_status',status_data)
+                        self.log.debug(f"write calibration data")
+                        self.writer.write('calibration', calib_data)
+
+    def finish(self):
+        Provenance().add_output_file(
+            self.output_file,
+            role='mon.tel.calibration'
+        )
+        self.writer.close()
+
+
+def main():
```

Comment:
How is this supposed to work?
Shouldn't a file be passed a some point?

Plus, I don't think we actually want a main here in the library but rather have a script in `scripts/`.

Reply:
This is just an example on how the flat filed and pedestal Components could be used in an event loop. This is a tool as conceived in ctapipe (for that I put it in the tool example)
As soon as the DL0-DL1 writing activity will be more advanced we will develop the executable that will perform the R0->R1->DL0/DL1 chain and write data on disk. This indeed should better go in  the /scripts directory. What do you think?

Reply:
Ok but it is just confusing (to me at least) to have a main here with an executable.
As a user/developer, I am tempted to run the file, without success.

Additionally, I couldn't test this piece of code. How should I proceed to do so?

Reply:
Why without success? 
You have an example on how to run it at the end of the notebook [notebooks/calculate_calibration_coefficients.ipynb](notebooks/calculate_calibration_coefficients.ipynb)

But if this drafted example is a sourced of confusion, I can take it away without problem.
I will create anyway an executable in the script directory.

Reply:
Because I am foolish and try to simply run `python calc_camera_calibration.py` as I see a `main()`.
Seeing the notebook, I understand now that a config and a file should be passed but the current script does not take any argument, right?

Up to me, a script with a `argparse` and a doc would be much more useful to a user :)

Reply:
> Seeing the notebook, I understand now that a config and a file should be passed but the current script does not take any argument, right?

No, it takes also some arguments. Try:
python calc_camera_calibration.py --help 

> Up to me, a script with a `argparse` and a doc would be much more useful to a user :)

actually I produced it following the Tools prescription of ctapipe
[https://cta-observatory.github.io/ctapipe/examples/Tools.html](https://cta-observatory.github.io/ctapipe/examples/Tools.html) that is the main way of using the Component classes, but I will have a look at the reco executables and produce an alternative way if better.

By the way, do you understand why travis flails now?

Reply:
Ok I see, I didn't quite understand how the Tool works sorry for that.
Maybe a very simple docstring stating `python calc_camera_calibration.py --help` in the main would be sufficient to avoid such misunderstanding to other users.

I think I don't have the required files to test the code though but that's fine if you tested it.
Thank you for the feedback and sorry for the mess.

Reply:
Yes, the problem of test files is unfortunately still not solved (the files that I have are all too big for uploading them) 
I added  a simple docstring.

Reply:
Thanks.
If you tested the code with the appropriate files I think we can merge. I approved the PR.